ale_linters/go/gobuild.vim

```diff
@@ -1,17 +1,126 @@
-" Author: dzhou121 <dzhou121@gmail.com>
+" Author: Joshua Rubin <joshua@rubixconsulting.com>
 " Description: go build for Go files
 
-function! s:FindGobuildScript() abort
-    return g:ale#util#stdin_wrapper . ' .go go build -o /dev/null'
+" inspired by work from dzhou121 <dzhou121@gmail.com>
+
+function! s:goenv()
+  if exists('s:go_env')
+    return s:go_env
+  endif
+
+  let l:env = systemlist('go env GOPATH GOROOT')
```

Comment:
You could put this in an initial command, too. You can write `'go env GOPATH GOROOT'` as the return value for a command callback, and you can return `''` from the callback to skip that command on the second call after that. Then the command will only be run once, but it won't block input in Vim on the first run, which is a little bit nicer. Take a look at the `flake8` linter for an example.